kivymd/uix/button/button.py

```diff
@@ -1272,12 +1273,13 @@ def __init__(self, **kwargs):
 
     def set_size(self, interval: Union[int, float]) -> None:
         """
-        Sets the custom icon size if the value of the `icon_size`
-        attribute is not zero. Otherwise, the icon size is set to `(48, 48)`.
+        Sets the icon width/height based on the current `icon_size`
+        attribute, or the default value if it is zero. The icon size
+        is set to `(48, 48)` for an icon with the default font_size 24sp.
         """
-
-        self.width = "48dp" if not self.icon_size else dp(self.icon_size + 23)
-        self.height = "48dp" if not self.icon_size else dp(self.icon_size + 23)
+        d = self._default_icon_pad + (self.icon_size or sp(24))
```

Comment:
Please do not use such variable naming. Each variable name should have a meaning.

Reply:
Fixed (d => diameter)